File: backend/app/schemas/oauth_token.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class OAuthToken(BaseModel):
    access_token: str
    refresh_token: str
    token_type: Optional[str] = "Bearer"
    expires_at: Optional[datetime] = None

    class Config:
        orm_mode = True

        model_config = {
        "from_attributes": True  # <-- allows conversion from ORM objects
    }

File: mobile/app/lib/api/models/oauth_token.dart
class OAuthToken {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final DateTime? expiresAt;

  OAuthToken({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    this.expiresAt,
  });

  factory OAuthToken.fromJson(Map<String, dynamic> json) {
    return OAuthToken(
      accessToken: json['access_token'],
      refreshToken: json['refresh_token'],
      tokenType: json['token_type'],
      expiresAt: json['expires_at'] != null
          ? DateTime.parse(json['expires_at'])
          : null,
    );
  }
}
